core/modules/installer/installer.browser.inc

```diff
@@ -0,0 +1,470 @@
+ *       'maintenance status' => 'Actively maintained', // Maintenance status
+ *       'development status' => 'Under active development', // Development status
+ *       'rating' => '9.6', // A rating on a scale of 1 to 10 of the project, if available
+ *       'dependencies' => array( // An array of the dependencies of this module
+ *         'entity',
+ *       ),
+ *     ),
+ *   'another_project => array( ... ),
+ *   ),
+ * );
+ *
+ * @param array $filters
+ *   An associative array of queries to use to filter results.
+ *
+ * @return
+ *   Returns an array of results.
+ */
+function installer_browser_fetch_results($filters) {
+  $server = installer_browser_get_server();
+  // Attempt to retrieve the cached version of this page.
+  $cid = md5(serialize(array_merge($filters, $server)));
+  if ($cache = cache_get($cid)) {
+    return $cache->data;
+  }
+
+  $results = array(
+    'projects' => array(),
+    'total' => 0,
+  );
+
+  $local_filters = $filters;
+
+  // Use XMLRPC if it is set.
+  if ($server['method'] == 'xmlrpc') {
+    $results = xmlrpc($server['url'], array(
+      'installer_browser_server.fetch_results' => array($local_filters),
+    ));
+
+    // Check for errors.
+    if ($error = xmlrpc_error() AND $error->is_error) {
+      backdrop_set_message(t("Encountered an error when trying to fetch results from @name. Error @code : @message",
+        array('@name' => $server['name'], '@code' => $error->code, '@message' => $error->message)));
+      return array();
+    }
+  }
+
+  // Use json if it is set.
+  if ($server['method'] == 'json') {
+    $local_filters['method'] = 'query';
+
+    $query_url = $server['url'] . '/query/' . $local_filters['type'] . '/1?' . http_build_query($local_filters, FALSE, '&');
+
+    $response = backdrop_http_request($query_url);
+
+    if ($response->code == '200') {
+      $results = backdrop_json_decode($response->data);
+    }
+    else {
+      backdrop_set_message(t("Encountered an error when trying to fetch results from @name. Error @code : @message",
+        array('@name' => $server['name'], '@code' => $response->code, '@message' => $response->error)));
+      return array();
+    }
+  }
+
+  // Set the cached version of the results.
+  cache_set($cid, $results, 'cache', strtotime("+24 hours"));
+
+  return $results;
+}
+
+/**
+ * Gets the server to use for fetching results.
+ *
+ * @return
+ *   Returns the active server, in the format:
+ *   array(
+ *     "url" => "https://projects.backdropcms.org",
+ *     "name" => "Backdrop",
+ *     "method" => "json"
+ *   )
+ */
+function installer_browser_get_server() {
+  $server = config_get('installer.settings', 'installer_browser_server');
+  backdrop_alter('installer_browser_server', $server);
+
+  return $server;
+}
+
+/**
+ * Uses the project status url to get the available releases for a project.
+ *
+ * @param $project
+ *   The project to get the releases for.
```

Comment:
Could use a better description here. What's in `$project` or where does it come from?